staker/rollup_watcher.go

```diff
@@ -48,12 +48,18 @@ type RollupWatcher struct {
 	*rollupgen.RollupUserLogic
 	address             common.Address
 	fromBlock           *big.Int
-	client              arbutil.L1Interface
+	client              RollupWatcherL1Interface
 	baseCallOpts        bind.CallOpts
 	unSupportedL3Method atomic.Bool
 }
 
-func NewRollupWatcher(address common.Address, client arbutil.L1Interface, callOpts bind.CallOpts) (*RollupWatcher, error) {
+type RollupWatcherL1Interface interface {
+	bind.ContractBackend
+	HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error)
+	FilterLogs(ctx context.Context, q ethereum.FilterQuery) ([]types.Log, error)
+}
```

Comment:
Here as well.

Reply:
The reasoning is how the `RollupWatcher` is created, previously `NewRollupWatcher` used to take a `arbutil.L1Interface` interface as an input and hence a `*ethclient.Client` was used in cmd/pruning while creating RollupWatcher and a `*txbuilder.Builder` was used in staker/l1validator, hence we substitute `arbutil.L1Interface` with `RollupWatcherL1Interface` to accommodate `*txbuilder.Builder`.

Similarly this also the reason for needing `ParentHeaderFetcher` interface.